src/util/Path.h

```diff
@@ -20,57 +20,49 @@ using std::string;
 class Path
 {
 public:
-	Path();
-	Path(const Path& other);
-	Path(string path);
-	Path(const char* path);
-	virtual ~Path();
+	Path() = default;
+	Path(const Path& other) = default;
+	Path(Path&& other) = default;
+
+	// Converting constructor it is intended not to use explicit there until c++17 (return {some_string}; will work).
+	Path(string path);       // NOLINT
+	Path(const char* path);  // NOLINT
+
+	~Path() = default;
+
 
 public:
 	/**
 	 * @return true if empty
 	 */
-	bool isEmpty();
+	bool isEmpty() const;
 
 	/**
 	 * Check if this file / folder exists
 	 */
-	bool exists();
+	bool exists() const;
 
 	/**
-	 * Delete the file
+ 	 * Delete the file
 	 *
-	 * @return true if the file is deleted or does not exists
+	 * @return true if the file existed and is deleted false if an error occurred
 	 */
 	bool deleteFile();
 
-	/**
-	 * Compare the path with another one
-	 */
-	bool operator ==(const Path& other);
+	bool operator==(const Path& other) const;
 
-	/**
-	 * Assign path
-	 */
-	void operator =(const Path& other);
-
-	/**
-	 * Assign path
-	 */
-	void operator =(const string& path);
-
-	/**
-	 * Assign path
-	 */
-	void operator =(const char* path);
+	Path& operator=(const Path& other) = default;
+	Path& operator=(Path&& other) = default;
+	Path& operator=(string path);
+	Path& operator=(const char* path);
 
 	/**
 	 * Check if the path ends with this extension
 	 *
 	 * @param ext Extension, needs to be lowercase
```

Comment:
Indicate in the documentation that `ext` must be prefixed with a `.` or else the extension check will not work correctly (it will only compare with the suffix of the path`).

Reply:
I just enabled the possibility to use both.